raft/minter.go

```diff
@@ -407,3 +424,29 @@ func (env *work) commitTransaction(tx *types.Transaction, bc *core.BlockChain, g
 
 	return publicReceipt, privateReceipt, nil
 }
+
+func (minter *minter) buildExtraSeal(headerHash common.Hash) []byte {
+	//Sign the headerHash
+	nodeKey := minter.eth.nodeKey
+	sig, err := crypto.Sign(headerHash.Bytes(), nodeKey)
+	if err != nil {
+		log.Warn("Block sealing failed", "err", err)
+	}
+
+	//build the extraSeal struct
+	raftIdString := hexutil.EncodeUint64(uint64(minter.eth.raftProtocolManager.raftId))
+
+	var extra *extraSeal
+	extra = &extraSeal{
+		RaftId: []byte(raftIdString[2:]), //remove the 0x prefix
+		Signature: sig,
+	}
+
+	//encode to byte array for storage
+	extraDataBytes, err := rlp.EncodeToBytes(&extra)
```

Comment:
`extra` is already a pointer, and we pass `&extra` to the RLP encode.
It might not be an issue as unit test pass but i think it's easier to understand to use `rlp.EncodeToBytes(extra)` instead.

Reply:
Good eye